WCFDataService/Client/System/Data/Services/Client/Materialization/EntryValueMaterializationPolicy.cs

```diff
@@ -510,7 +510,11 @@ private object GetOrCreateCollectionProperty(object instance, ClientPropertyAnno
                         collectionType = typeof(System.Collections.ObjectModel.Collection<>).MakeGenericType(property.EntityCollectionItemType);
                     }
 
-                    result = this.CreateNewInstance(property.EdmProperty.Type, collectionType);
+                    // Support proxies with uninitialized DataServiceCollection<> type properties.
```

Comment:
This path will be hitted when result = property.GetValue(instance) and it is not LoadProperty. When will "result = property.GetValue(instance)" returns null. For undeclared property?